docs/solutions/sql_server_always_on_availability_groups/modules/network/secondaryip/main.tf

```diff
@@ -0,0 +1,17 @@
+resource "oci_core_private_ip" "ip2" {
+  // Terraform sometimes can be stubborn and doesn't like referencing datasource  // as a count value. We're going to use ad_count value to count the number of hosts  // For larger number of hosts we could introduce a variable ${var.hosts_num} and use  // math interpolation to calculate number of additional addresses.
```

Comment:
I've gotten into a bad state while trying to apply this config. The initial apply failed when I hit limits on the number of instances, and so a subnet of the instances were deployed. Now when I run plan, apply, or destroy, it fails with the following:
  Error: Error refreshing state: 2 error(s) occurred:
   * module.secondaryIPs.oci_core_private_ip.ip3: 1 error(s) occurred:
   * module.secondaryIPs.oci_core_private_ip.ip3[2]: index 2 out of range for list var.vnic_ids (max 2) in:
   ${lookup(var.vnic_ids["${count.index}"],"vnic_id")}
   * module.secondaryIPs.oci_core_private_ip.ip2: 1 error(s) occurred:
   * module.secondaryIPs.oci_core_private_ip.ip2[2]: index 2 out of range for list var.vnic_ids (max 2) in:
   ${lookup(var.vnic_ids["${count.index}"],"vnic_id")}

Looking for a solution...

Reply:
After a lot of trial and error, I came up with:

  vnic_id = "${lookup(var.vnic_ids[count.index < length(var.vnic_ids) ? count.index : 0] ,"vnic_id")}"

That's pretty ugly and not perfect, but it did get me out of the bad state I was in, and I verified that it works correctly on a fresh deployment as well. So I would recommend switching to this both here and in ip3 a few lines down.

Reply:
Not sure how common it would be to get into the situation that I did (it seems to happen if the apply fails with a partial deployment of these resources), but it's pretty bad if you do get into this state. It would be great to get the change that I'm suggesting in for the initial release, and if not then shortly after. This change would require retesting the E2E scenario to make sure it's correct.

Reply:
We will add this to the refactor list.  I agree that this can be an issue if you run into limits, but I would also agree that I am not sure that this will be reached often and should be treated as a corner case.